src/components/common/CategoryFilter/index.tsx

```diff
@@ -0,0 +1,56 @@
+import { ButtonBase, Grid, IconButton, Typography } from '@mui/material'
+import { useRouter } from 'next/router'
+import clsx from 'clsx'
+import CloseIcon from '@/public/images/close.svg'
+import css from './styles.module.css'
+
+const CategoryFilter = ({ categories }: { categories: string[] }) => {
+  const router = useRouter()
+  const selectedCategory = router.query.category
+
+  const handleToggleCategory = (category: string) => () => {
+    const queryParams = { ...router.query }
+
+    if (queryParams.category === category) {
+      delete queryParams.category
+    } else {
+      queryParams.category = category
+    }
+
+    router.push(
+      {
+        query: queryParams,
+      },
+      undefined,
+      { shallow: true },
+    )
+  }
+
+  return (
+    <Grid container className={css.filterWrapper}>
+      {categories.map((category) => {
+        const isSelected = category === selectedCategory
+
+        return (
+          <Grid item key={category} className={css.filterCard} xs={12} md="auto">
+            <ButtonBase
+              className={clsx(css.filterButton, { [css.selected]: isSelected })}
+              onClick={handleToggleCategory(category)}
+            >
+              <Typography>
+                {category}
+                {isSelected && (
+                  <IconButton className={css.closeFilter} onClick={handleToggleCategory(category)}>
```

Comment:
Currying this still creates the function twice. You can create it within the `map` once, e.g.
```suggestion
  const handleToggleCategory = (category: string) => {
    const queryParams = { ...router.query }

    if (queryParams.category === category) {
      delete queryParams.category
    } else {
      queryParams.category = category
    }

    router.push(
      {
        query: queryParams,
      },
      undefined,
      { shallow: true },
    )
  }

  return (
    <Grid container className={css.filterWrapper}>
      {categories.map((category) => {
        const isSelected = category === selectedCategory
        const handleToggle = () => handleToggleCategory(category)

        return (
          <Grid item key={category} className={css.filterCard} xs={12} md="auto">
            <ButtonBase
              className={clsx(css.filterButton, { [css.selected]: isSelected })}
              onClick={handleToggle}
            >
              <Typography>
                {category}
                {isSelected && (
                  <IconButton className={css.closeFilter} onClick={handleToggle}>
```

Reply:
WDYT of memoizing the function with `useCallback` ?

Reply:
Moved the handler inside of the mapping function